pkg/cmd/repo/garden/garden.go

```diff
@@ -0,0 +1,412 @@
+	case DirRight:
+		if p.X == p.Geo.Width-1 {
+			return
+		}
+		p.X++
+	}
+}
+
+type GardenOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	RepoArg string
+}
+
+func NewCmdGarden(f *cmdutil.Factory, runF func(*GardenOptions) error) *cobra.Command {
+	opts := GardenOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:    "garden [<repository>]",
+		Short:  "Explore a git repository as a garden",
+		Long:   "Use WASD or vi keys to move. q to quit.",
+		Hidden: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return gardenRun(&opts)
+		},
+	}
+
+	return cmd
+}
+
+func gardenRun(opts *GardenOptions) error {
+	out := opts.IO.Out
+
+	if runtime.GOOS == "windows" {
+		return errors.New("sorry :( this command only works on linux and macos")
+	}
+
+	if !opts.IO.IsStdoutTTY() {
+		return errors.New("must be connected to a terminal")
+	}
+
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	var toView ghrepo.Interface
+	apiClient := api.NewClientFromHTTP(httpClient)
+	if opts.RepoArg == "" {
+		var err error
+		toView, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+	} else {
+		var err error
+		viewURL := opts.RepoArg
+		if !strings.Contains(viewURL, "/") {
+			currentUser, err := api.CurrentLoginName(apiClient, ghinstance.Default())
+			if err != nil {
+				return err
+			}
+			viewURL = currentUser + "/" + viewURL
+		}
+		toView, err = ghrepo.FromFullName(viewURL)
+		if err != nil {
+			return fmt.Errorf("argument error: %w", err)
+		}
+	}
+
+	seed := computeSeed(ghrepo.FullName(toView))
+	rand.Seed(seed)
+
+	termWidth, termHeight, err := utils.TerminalSize(out)
+	if err != nil {
+		return err
+	}
+
+	termWidth -= 10
+	termHeight -= 10
+
+	geo := &Geometry{
+		Width:      termWidth,
+		Height:     termHeight,
+		Repository: toView,
+		// TODO based on number of commits/cells instead of just hardcoding
+		Density: 0.3,
+	}
+
+	maxCommits := geo.Width * geo.Height
+
+	commits, err := getCommits(httpClient, toView, maxCommits)
+	if err != nil {
+		return err
+	}
+	player := &Player{0, 0, utils.Bold("@"), geo, 0}
+
+	garden := plantGarden(commits, geo)
+	clear(opts.IO)
+	drawGarden(out, garden, player)
+
+	// thanks stackoverflow https://stackoverflow.com/a/17278776
+	if runtime.GOOS == "darwin" {
+		_ = exec.Command("stty", "-f", "/dev/tty", "cbreak", "min", "1").Run()
+		_ = exec.Command("stty", "-f", "/dev/tty", "-echo").Run()
+	} else {
+		_ = exec.Command("stty", "-F", "/dev/tty", "cbreak", "min", "1").Run()
+		_ = exec.Command("stty", "-F", "/dev/tty", "-echo").Run()
+	}
+
+	var b []byte = make([]byte, 1)
+	for {
+		_, err := opts.IO.In.Read(b)
+		if err != nil {
+			return err
+		}
+
+		quitting := false
+		switch {
```

Comment:
Could we add a `default:` case to this switch statement that just skips to the next loop iteration without redrawing the garden? That would avoid re-rendering the garden on invalid keystrokes such as arrow keys